sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/LinuxKeyRingAccessor.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.implementation;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.aad.msal4jextensions.persistence.linux.ISecurityLibrary;
+import com.sun.jna.Pointer;
+import com.sun.jna.Structure;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class exposes functions from Key Ring on Linux platform
+ * via JNA.
+ */
+public class LinuxKeyRingAccessor {
```

Comment:
Looks like we're using msal extension library's `KeyChainAccessor` class  to access KeyChain on mac. Is no similar class available for Key Ring on linux?

Reply:
So, 
MSAL's implementation Base 64 encodes the secret/credential when writing to Key Ring.
&& Base 64 decodes when reading from Key Ring.
But, IntelliJ + Vs code don't base 64 encode the credential details when writing to Key Ring, so it MSAL's implementation blows up when reading it.